Client/Ecosia/Braze/BrazeService.swift

```diff
@@ -0,0 +1,161 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import BrazeKit
+import BrazeUI
+import Core
+
+final class BrazeService {
+    private init() {}
+
+    private var braze: Braze?
+    private var userId: String {
+        User.shared.analyticsId.uuidString
+    }
+    private(set) var notificationAuthorizationStatus: UNAuthorizationStatus?
+    private static var apiKey = EnvironmentFetcher.valueFromMainBundleOrProcessInfo(forKey: "BRAZE_API_KEY") ?? ""
+    static let shared = BrazeService()
+
+    enum Error: Swift.Error {
+        case invalidConfiguration
+        case generic(description: String)
+    }
+
+    enum CustomEvent: String {
+        case newsletterCardClick = "newsletter_card_click"
+    }
+
+    // TODO: Make `BrazeService` directly `UNUserNotificationCenterDelegate`
```

Comment:
@d4r1091 do you see any issue with that and reason for it to remain as the `AppDelegate`? Imo it would remove the need to pass this as parameter to some places and make the code cleaner.